src/lib.rs

```diff
@@ -250,18 +251,33 @@ impl<'a> Decompressor<'a> {
         Self { symbols, lengths }
     }
 
+    /// Returns the capacity required for decompression.
+    pub fn decompressed_capacity(&self, compressed: &[u8]) -> usize {
+        size_of::<Symbol>() * (compressed.len() + 1)
+    }
```

Comment:
we should call this `max_decompression_capacity`. it's an upper-bound since symbols need not be 8 bytes (and most of them time they'll be <= 3bytes)

Reply:
Yep fair, although the check is that the slice must be == this capacity. We could make it <=, but would require a bounds check inside each iteration which feels... slow